### ISSUE TITLE
Image inside SVG don't  get rendered

### ISSUE THREAD

**Issue by hbroer** [wkhtmltopdf/wkhtmltopdf#2996]
Hey,

i have a problem to include a SVG File which itself includes images. The Images are not visible inside the PDF output. I have that Problem on a Ubuntu 14.04 Server and also on my local workstation (Manjaro).

html File
`<img class="background" src="svgfile.svg" alt="" />`

svg File
```
<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 595.3 841.9">
[....otherelements...]
<image  width="507" height="440" xlink:href="data:image/jpeg;base64,[MUCHDATA]" ></image>
[....otherelements...]
</svg>
```

If i open that SVG in browser (Firefox/Chrome) everything is ok. I also tried to include that svg with embed and object tag, but then that PDF creation fails completely. If i include the SVG directly into that HTML File the Images are visible (but some other elements are not rendered properly.

I hope someone can help me to find a workaround, or knows how to fix that.

wkhtmltopdf 0.12.3 (with patched qt)

**Comment by jakeplaras** [comment]
@hbroer  hey man... I'm having the same problem... did you find a solution for this? I tried encoding to base64... it didn't help.

**Comment by hbroer** [comment]
@jakeplaras no, i didn't find a workaround. We place all images now directly into HTML and include all vector images directly to the SVG. That isn't the best solution, because we use svg images as drafts in the background, and the imposibility to embed images is very limiting.

Before wie usw wkhtmltopdf we load a PDF as draft and add the new elements. But i couldn't figure out how to embed PDF, i think that isn't possible.

Base64 to embed images into svg didn't work here too.

**Comment by jakeplaras** [comment]
Thanks for the quick response @hbroer .. I'm going to try to find a workaround for this. I'll post it here if I get lucky.

**Comment by zurez** [comment]
Any luck? I have same issue.
PS: using object tag leads to segmentation error

@jakeplaras @hbroer

**Comment by PhilterPaper** [comment]
Is this the same problem as #1951?

**Comment by zurez** [comment]
Similar not Same.

**Comment by ctwheels** [comment]
I had issues with this as well. the `image` tag within `svg` do not get rendered correctly.
To fix this, I created a nested `svg` with the proper sizing for the image containing a `foreignobject`, which in turn holds an `img`.
The reason for the nested `svg` is to prevent odd displacement of sibling `svg` elements, which occurred only in wkhtmltopdf and not in a typical browser like Chrome

This is an example:
```
<svg x="0" y="0" width="100%" height="100%">
  <svg x="0" y="0" width="100%" height="100%">
    <foreignobject width="100%" height="100%">
      <img width="100%" src="https://upload.wikimedia.org/wikipedia/commons/a/a0/SG100344.JPG" />
    </foreignobject>
  </svg>
</svg>
```

using this CSS, the image is rendered on the full page (if you're wondering why only a section shows up)
```
html, body {
  padding: 0;
  margin: 0;
  width :100%;
  height: 100%;
  overflow: hidden;
}
```

**Comment by ppant** [comment]
I am also facing similar issue. My html files has code like:
<p><object id="SVGViewer1" data="visio_test_svg.svg"  WIDTH=2500 HEIGHT=3000 type="image/svg+xml" VIEWASTEXT></object></p>

PDF output is blank. Direct svg -> pdf creation is fine.

Using v0.12.3 (patched qt)

**Comment by sudiptokarmoker** [comment]
Its really a pathetic situation. because image tag in svg works on chrome. but its not on firefox. it works fine before some days. but currently its not being showed. If anyone could help here....

**Comment by ticapix** [comment]
Same with `wkhtmltopdf 0.12.4 (with patched qt)`, the SVG file is not rendered

```html
	<div class="logo">
		<img src="logo.svg">
	</div>
```

**Comment by CardBook** [comment]
same problem for me...

**Comment by viswa18** [comment]
same issue here.. Is there any option to overcome this?

**Comment by RecursiveProgrammer** [comment]
Greetings all,

Not to hijack the thread (I have an issue where embedding an SVG in an html document is rendering a blank page, but my html code is a bit different), but I have a question for @alloylab : It appears that this topic is flagged as needing info, but no question was asked (it seems), by you. Could you specify which information that you are waiting for? 

Cheers!

**Comment by Tomsgu** [comment]
Does this problem still appears in 0.12.5 version? If does can you post a minimal example so the issue can be reproduced?

**Comment by martin-frydl** [comment]
The problem still exists in 0.12.5 on Windows. However, it does work on Linux. 

- Windows --version: wkhtmltopdf 0.12.5 (with patched qt)
- Linux --version: wkhtmltopdf 0.12.5 -> comes from wkhtmltopdf-0.12.5-1.fc27.x86_64 (Fedora 27)
- Windows "bleeding edge" also fails: wkhtmltopdf 0.13.0-alpha-7b36694

[test.zip](https://github.com/wkhtmltopdf/wkhtmltopdf/files/2845279/test.zip)

**Comment by yinyang8** [comment]
@ticapix - Is there a way to get the unpatched version of 0.12.2.4 and/or does 0.12.2.3 also work?  Thanks.

**Comment by partychen** [comment]
Is there some updates?

**Comment by glassdimly** [comment]
Use the `image` tag instead: SVGTag({speed, theme})

**Comment by leonelvsc** [comment]
I ran into the same problem and resolved it by adding the embeded img to the main html context but with a display style of none ie of this post will be like: 

**original:**
```
<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 595.3 841.9">
[....otherelements...]
<image  width="507" height="440" xlink:href="data:image/jpeg;base64,[MUCHDATA]" ></image>
[....otherelements...]
</svg>
```

**possible fix:**
html file:
```
<img style="display: none"  width="507" height="440" src="data:image/jpeg;base64,[MUCHDATA]" >
<img src="[svg-file]">
```

svg file:
```
<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 595.3 841.9">
[....otherelements...]
<image  width="507" height="440" xlink:href="data:image/jpeg;base64,[MUCHDATA]" ></image>
[....otherelements...]
</svg>
```

i think it is because the render engine can declare the data uri from the image but not from the embedded image inside the svg image

**Comment by dimonx3** [comment]
when are you planning to fix it?

**Comment by ticapix** [comment]
Hello, I provided a quick&dirty solution in https://github.com/Kozea/WeasyPrint/issues/1196#issuecomment-857898964

I hope you'll find it useful.

In the meantime, I also came across draw.io that has a [CLI](https://github.com/rlespinasse/drawio-cli) and can export SVG with `<foreignObject>` without issue. However it needs to wrapper around it to `.svg` -> `.drawio` -> `.png`

**Comment by jeronimoek** [comment]
> I had issues with this as well. the `image` tag within `svg` do not get rendered correctly. To fix this, I created a nested `svg` with the proper sizing for the image containing a `foreignobject`, which in turn holds an `img`. The reason for the nested `svg` is to prevent odd displacement of sibling `svg` elements, which occurred only in wkhtmltopdf and not in a typical browser like Chrome
> 
> This is an example:
> 
> ```
> <svg x="0" y="0" width="100%" height="100%">
>   <svg x="0" y="0" width="100%" height="100%">
>     <foreignobject width="100%" height="100%">
>       <img width="100%" src="https://upload.wikimedia.org/wikipedia/commons/a/a0/SG100344.JPG" />
>     </foreignobject>
>   </svg>
> </svg>
> ```

This worked for me, TY so much!! I used it with a base64 image
